vmdb/app/assets/javascripts/controllers/provider_foreman/provider_foreman_form_controller.js

```diff
@@ -0,0 +1,102 @@
+miqAngularApplication.controller('providerForemanFormController', ['$http', '$scope', 'providerForemanFormId', 'miqService', function($http, $scope, providerForemanFormId, miqService) {
+    var init = function() {
+      $scope.providerForemanModel = {
+        name: '',
+        url: '',
+        verify_ssl: '',
+        log_userid: '',
+        log_password: '',
+        log_verify: ''
+      };
+      $scope.formId = providerForemanFormId;
+      $scope.afterGet = false;
+      $scope.modelCopy = angular.copy( $scope.providerForemanModel );
+
+      miqAngularApplication.$scope = $scope;
+
+      if (providerForemanFormId == 'new') {
+        $scope.newRecord                         = true;
+        $scope.providerForemanModel.name         = '';
+        $scope.providerForemanModel.url          = '';
+        $scope.providerForemanModel.verify_ssl   = '';
+
+        $scope.providerForemanModel.log_userid   = '';
+        $scope.providerForemanModel.log_password = '';
+        $scope.providerForemanModel.log_verify   = '';
+        $scope.afterGet                          = true;
+        $scope.modelCopy                         = angular.copy( $scope.providerForemanModel );
+      } else {
+        $scope.newRecord = false;
+
+        miqService.sparkleOn();
+
+        $http.get('/provider_foreman/provider_foreman_form_fields/' + providerForemanFormId).success(function(data) {
```

Comment:
This URL shouldn't be hardcoded.  We should put it in a `data-` element on the page and get it with JS.